src/SDK/Resource/Detectors/Host.php

```diff
@@ -15,13 +15,100 @@
  */
 final class Host implements ResourceDetectorInterface
 {
+    private const PATH_ETC_MACHINEID = 'etc/machine-id';
+    private const PATH_VAR_LIB_DBUS_MACHINEID = 'var/lib/dbus/machine-id';
+    private const PATH_ETC_HOSTID = 'etc/hostid';
+    private readonly string $dir;
+    private readonly string $os;
+
+    public function __construct(string $dir = '/', string $os = PHP_OS_FAMILY)
+    {
+        $this->dir = $dir;
+        $this->os = $os;
+    }
+
     public function getResource(): ResourceInfo
     {
         $attributes = [
             ResourceAttributes::HOST_NAME => php_uname('n'),
             ResourceAttributes::HOST_ARCH => php_uname('m'),
+            ResourceAttributes::HOST_ID => $this->getMachineId(),
         ];
 
         return ResourceInfo::create(Attributes::create($attributes), ResourceAttributes::SCHEMA_URL);
     }
+
+    private function getMachineId(): ?string
+    {
+        switch ($this->os) {
+            case 'Linux':
+                {
+                    return $this->getLinuxId();
+                }
+            case 'BSD':
+                {
+                    return $this->getBsdId();
+                }
+            case 'Darwin':
+                {
+                    return $this->getMacOsId();
+                }
+            case 'Windows':
+                {
+                    return $this->getWindowsId();
+                }
+        }
+
+        return null;
```

Comment:
Potentially a candidate for using a match statement here:
```suggestion
        return match ($this->os) {
            'Linux' => $this->getLinuxId(),
            'BSD' => $this->getBsdId(),
            'Darwin' => $this->getMacOsId(),
            'Windows' => $this->getWindowsId(),
            default => null,
        };
```